src/reflect-shim.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+ /**
+ * This shim allows elements written in, or compiled to, ES5 to work on native
+ * implementations of Custom Elements v1. It sets new.target to the value of
+ * this.constructor so that the native HTMLElement constructor can access the
+ * current under-construction element's definition.
+ *
+ * Because `new.target` is a syntax error in VMs that don't support it, this
+ * shim must only be loaded in browsers that do.
+ */
+(() => {
+  if (window.Reflect === undefined) {
+    // No Reflect, no classes, no need for shim
+    return;
+  }
+  const BuiltInHTMLElement = HTMLElement;
+  window.HTMLElement = function() {
+    return Reflect.construct(BuiltInHTMLElement, [], this.constructor);
```

Comment:
`this.constructor` -> `new.target`

Reply:
Oh, nevermind, this isn't set because we're assuming they've used `call` to get here... *facepalm*

Reply:
That, but mainly because `new.target` is a syntax error in older browsers.